src/modules/claims/claims.service.ts

```diff
@@ -791,18 +804,6 @@ export class ClaimsService {
     // can we break API so that register on chain required only claim type and claim type version and subject
     if (registrationTypes.includes(RegistrationTypes.OffChain)) {
       const token = claim.token as string;
-      const payload = (await this._didRegistry.decodeJWTToken({ token })) as {
```

Comment:
Do you think we should check if subject is current DID?

Reply:
Does the behavior change if it is one vs the other? @JGiter

Reply:
It is just a matter of error handling. Currently did-registry doesn't handles failed update of document. Debugging would be easer if iam-lib prevent such error with explanatory message

Reply:
But I don't think it should be an error if the subject isn't the current DID. In the case of an asset it wouldn't be. Maybe I'm not following what you are saying though.

Reply:
Ah, right. `updateDocument` checks if document is controlled